doc/concept/core.md

```diff
@@ -0,0 +1,7 @@
+# core
+The `core` is the implementation of Anna's neural networks. The core bundles
+everything around data processing and intelligence. It provides `gateways` were
+i/o is goint through. The core itself can contain multiple `networks`. Network
+implementations provide `neurons` that interact using `connections`.
```

Comment:
The concepts here need more explanation. Since this can only be done as soon as the concept is clear, we might not document everything right now. Anyway, the `gateway` and the `server` could already be explained.